cpp/inline/games/nim/Game.inl

```diff
@@ -0,0 +1,98 @@
+#include <games/nim/Game.hpp>
+
+namespace nim {
+
+inline void Game::Rules::init_state(State& state) {
+  state.stones_left = kStartingStones;
+  state.current_player = 0;
+}
+
+inline size_t Game::State::hash() const {
+  auto tuple = std::make_tuple(stones_left, current_player, chance_active);
+  std::hash<decltype(tuple)> hasher;
+  return hasher(tuple);
+}
+
+inline Game::Types::ActionMask Game::Rules::get_legal_moves(const StateHistory& history) {
+  const State& state = history.current();
+  Types::ActionMask mask;
+  bool is_chance = is_chance_mode(get_action_mode(history.current()));
```

Comment:
This is unnecessary. Just directly use `state.chance_active`. That's kind of the whole point of making those members public.